frontend/src/metabase/query_builder/components/view/ViewHeader/components/QuestionActions/QuestionActions.tsx

```diff
@@ -168,15 +177,26 @@ export const QuestionActions = ({
     }
   }
 
-  if (isQuestion || isMetric) {
-    extraButtons.push({
-      title: t`Add to dashboard`,
-      icon: "add_to_dash",
-      action: () => onOpenModal(MODAL_TYPES.ADD_TO_DASHBOARD),
-      testId: ADD_TO_DASH_TESTID,
-    });
+  if (hasCollectionPermissions) {
+    if (isQuestion) {
+      extraButtons.push({
+        title: t`Turn into a model`,
+        icon: "model",
+        action: handleTurnToModel,
+        testId: TURN_INTO_DATASET_TESTID,
+      });
+    }
+    if (isModel) {
+      extraButtons.push({
+        title: t`Turn back to saved question`,
```

Comment:
(thinking out loud:) It's kind of a shame that this "turn back" language got used, since when you upload a CSV, it _starts out_ as a model.